File: software/examples/javascript/ExampleOutput.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your IO-16 Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var io = new Tinkerforge.BrickletIO16V2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Configure channel 7 as output low
        io.setConfiguration(7, 'o', false);

        // Set channel 7 alternating high/low 10 times with 100 ms delay
        for(var i = 0; i < 10; ++i) {
            setTimeout(function () {
                io.setSelectedValue(7, true);
            }, 200 * i + 100);
            setTimeout(function () {
                io.setSelectedValue(7, false);
            }, 200 * i + 200);
        }
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/java/ExampleOutput.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletIO16V2;

public class ExampleOutput {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your IO-16 Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIO16V2 io = new BrickletIO16V2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Configure channel 7 [A7] as output low
		io.setConfiguration(7, 'o', false);

		// Set channel 7 [A7] alternating high/low 10 times with 100 ms delay
		for(int i = 0; i < 10; i++) {
			Thread.sleep(100);
			io.setSelectedValue(7, true);
			Thread.sleep(100);
			io.setSelectedValue(7, false);
		}

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
